File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/LengthMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Length
import com.phamtunglam.health_connector_hc_android.pigeon.LengthDto

/**
 * Converts a [LengthDto] to a Health Connect [Length] object.
 */
internal fun LengthDto.toHealthConnect(): Length = Length.meters(meters)

/**
 * Converts a Health Connect [Length] object to a [LengthDto].
 *
 * Uses meters as the transfer unit for consistency.
 */
internal fun Length.toDto(): LengthDto = LengthDto(
    meters = inMeters,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/length_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Length;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show LengthDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Length] to [LengthDto].
@internal
extension LengthToDto on Length {
  LengthDto toDto() {
    // Uses meters as the transfer unit for consistency.
    return LengthDto(meters: inMeters);
  }
}

/// Converts [LengthDto] to [Length].
@internal
extension LengthDtoToDomain on LengthDto {
  Length toDomain() {
    return Length.meters(meters);
  }
}
